### PR TITLE
Skip communication manager thread for requests.

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR
@geirst FYI

### CODE REVIEW

File: storage/src/vespa/storage/common/storagelink.cpp [hunk 1]
```diff
@@ -165,9 +165,9 @@ void StorageLink::sendUp(const shared_ptr<StorageMessage> & msg)
         default:
             LOG(error, "Link %s trying to send %s up while in state %s",
                 toString().c_str(), msg->toString(true).c_str(), stateToString(getState()));
-            assert(false);
+            return;
```
Comment (vekterli) [inline storage/src/vespa/storage/common/storagelink.cpp:168]: Is it safe to remove these abort-branches? If the link is not in a state where it's safe to process certain types of messages, the behavior is likely undefined

File: messagebus/src/vespa/messagebus/messagebus.cpp [hunk 2]
```diff
@@ -112,7 +112,7 @@ MessageBus::MessageBus(INetwork &net, const MessageBusParams &params) :
     _routingTables(),
     _sessions(),
     _protocolRepository(std::make_unique<ProtocolRepository>()),
-    _msn(std::make_unique<Messenger>()),
+    _msn(std::make_unique<Messenger>(true, true)),
```
Comment (vekterli) [inline messagebus/src/vespa/messagebus/messagebus.cpp:115]: Is it currently safe to have thread-skipping be the default behavior for this constructor overload? Or should this be `false, false` instead?

**Review by vekterli — APPROVED** [review]
:+1: a couple of questions that would be good to resolve prior to merging

File: storage/src/vespa/storage/storageserver/communicationmanager.cpp [hunk 3]
```diff
@@ -437,6 +443,18 @@ CommunicationManager::process(const std::shared_ptr<api::StorageMessage>& msg)
     }
 }
 
+void
+CommunicationManager::optionalEnqueue(std::shared_ptr<api::StorageMessage> msg)
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/storageserver/communicationmanager.cpp]: Fixed

File: messagebus/src/vespa/messagebus/messagebus.cpp (hunk 2 above)
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/messagebus.cpp:115]: This constructor is only used by tests. My thought is that everything except distributor and proton should run with future defaults.

File: storage/src/vespa/storage/common/storagelink.cpp (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/common/storagelink.cpp]: It was initially removed, as there was a test failing in communication manager. However there is no longer need for this as the test itself was fixed.

**Pull request merged**